src/main/java/Task.java

```diff
@@ -0,0 +1,36 @@
+
+/**
+ * Encapsulates a Task.
+ */
+class Task {
+    /**
+     * Status of the Task.
+     */
+    boolean completed = false;
+
+    /**
+     * Details of the Task.
+     */
+    String task;
+
+    public Task(String task) {
+        this.task = task;
+    }
+
+    /**
+     * Provides Details of the Task.
+     * @return String detail message of Task.
+     */
+    public String provideDetails() {
+        return completed ? "[x] " + task
+                : "[ ] " + task;
+    }
+
+    /**
+     * Flips the status of the Task.
+     */
+    public void updateTask() {
```

Comment:
If the method is to flip the status, can we change the method name to `toggleStatus` to make it more self explanatory?